numba/cuda/cudaimpl.py

```diff
@@ -627,27 +628,43 @@ def ptx_min_f8(context, builder, sig, args):
     ])
 
 
-@lower(round, types.f4)
-@lower(round, types.f8)
-def ptx_round(context, builder, sig, args):
-    fn = cgutils.get_or_insert_function(
-        builder.module,
-        ir.FunctionType(
-            ir.IntType(64),
-            (ir.DoubleType(),)),
-        '__nv_llrint')
-    return builder.call(fn, [
-        context.cast(builder, args[0], sig.args[0], types.double),
-    ])
+# @lower(round, types.f4)
+# @lower(round, types.f8)
+@intrinsic
+def ptx_round(typingctx, x):
+    sig = types.int64(x)
+
+    def codegen(context, builder, sig, args):
+        fn = cgutils.get_or_insert_function(
+            builder.module,
+            ir.FunctionType(
+                ir.IntType(64),
+                (ir.DoubleType(),)),
+            '__nv_llrint')
+        return builder.call(fn, [
+            context.cast(builder, args[0], sig.args[0], types.double),
+        ])
+
+    return sig, codegen
 
 
 # This rounding implementation follows the algorithm used in the "fallback
 # version" of double_round in CPython.
 # https://github.com/python/cpython/blob/a755410e054e1e2390de5830befc08fe80706c66/Objects/floatobject.c#L964-L1007
+@overload(round)
+def round_to_impl(x):
+    if isinstance(x, types.Float):
+        def impl(x):
+            return ptx_round(x)
+        return impl
+
+
+@overload(round)
+def round_to_impl2(x, ndigits):
+    # XXX: it doesn't look like the CUDA target supports ndigits=None
```

Comment:
Can it support `ndigits=None` with something like:
```python
    if isinstance(ndigits, NoneType):
        def impl(x):
            return ptx_round(x)
        return impl
```
?

Reply:
I think there's at least two problems going on here, one of which is #8528, the other is what is causing this lowering error. It seems like non-matching default kwargs might be part of the cause, it's also strange that the overload validation step doesn't catch this.

Reply:
This is also involved: #8530